Readability.js

```diff
@@ -624,6 +624,11 @@ Readability.prototype = {
 
     var pageCacheHtml = page.innerHTML;
 
+    var articleTags = this._doc.getElementsByTagName("article");
+    if (articleTags.length === 1) {
+      this._doc.body.innerHTML = articleTags[0].parentNode.innerHTML;
```

Comment:
There are some issues here.

1. This will break if the page goes through the `while` loop below a second time, as we'll re-set the `innerHTML` to `pageCacheHtml` which is set directly above this block.
2. This leaves a whole lot of work inside the code here, like looking for candidates and whatnot, whereas the summary seems to indicate we just want to use the entire contents of the `article` tag. Why do it this way, then?
3. Why the `parentNode` of the `article`, which will include siblings?
4. This has a performance penalty: we'll be reparsing the HTML. I would suggest:
`this._doc.body.parentNode.replaceChild(articleTags[0].parentNode, this._doc.body)` which is almost O(1).